### PR TITLE
Meetups/list search sort

### PR DESCRIPTION
**Pull request by atappy**
I finally managed to switch to a recyclerview instead of a scrollView, and added a search button to search an activity by name. 

I also created a Filter class in util, to centralize the filter in one unique class, and added a way to test easily a RecyclerView (the class RecyclerViewMatcher)

### CODE REVIEW

File: app/src/main/java/com/github/palFinderTeam/palfinder/utils/Filter.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.github.palFinderTeam.palfinder.utils
+
+import android.widget.Adapter
+import android.widget.Filter
+import android.widget.SearchView
+import com.github.palFinderTeam.palfinder.meetups.MeetUp
+import com.github.palFinderTeam.palfinder.tag.Tag
+import java.util.*
+
+    public class searchedFilter<T>(private val dataSet : List<T>, private val currentDataSet : MutableList<T>, private val callBack : () -> Unit): Filter() {
```
Comment (zaclegarssure) [inline app/src/main/java/com/github/palFinderTeam/palfinder/utils/Filter.kt:10]: It should start with a capital letter (the classname)

**Review by zaclegarssure — APPROVED** [review]
Otherwise everything looks great.

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/MeetupListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.meetup_listview, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        //update displayed elements with the content of the current dataset
+        val meetup_title = holder.meetup_title
+        meetup_title.text = currentDataSet[position].name
+        val meetup_date = holder.meetup_date
+        val prettyDate = PrettyDate()
+        meetup_date.text = "in" + prettyDate.timeSince(currentDataSet[position].startDate.time)
```
Comment (TruddyTheDuddi) [inline app/src/main/java/com/github/palFinderTeam/palfinder/meetups/MeetupListAdapter.kt:39]: Check the new version of PrettyDate, it works now with forward dates. Also no need to use the prefix "in", "in" and "ago" likewise are generated automatically

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetupListActivity.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.*
+
+
+class MeetupListActivity : AppCompatActivity() {
+    private lateinit  var meetupList : RecyclerView
+    private lateinit var listOfMeetup : List<MeetUp>
+    private lateinit var adapter: MeetupListAdapter<MeetUp>
+
+
+    @RequiresApi(Build.VERSION_CODES.N)
```
[the remark this answers is not in the document]
Comment (atappy) [reply app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetupListActivity.kt:26]: I remove it

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/MeetupListAdapter.kt (hunk 2 above)
Comment (atappy) [reply app/src/main/java/com/github/palFinderTeam/palfinder/meetups/MeetupListAdapter.kt:39]: I think the version I have is not the newest, so I will update it as soon as possible

File: app/src/main/java/com/github/palFinderTeam/palfinder/utils/Filter.kt (hunk 1 above)
Comment (atappy) [reply app/src/main/java/com/github/palFinderTeam/palfinder/utils/Filter.kt:10]: fixed

**Comment by zaclegarssure** [comment]
oups

**Pull request merged**